examples/DivisorCurve.jl

```diff
@@ -0,0 +1,327 @@
+    return ProjCurveDivisor(C, Dict(P => m))
+end
+
+################################################################################
+# To get a nice output for the divisors.
+
+function AbstractAlgebra.needs_parentheses(A::Array{T, 1}) where T <: RingElem
+    return false
+end
+
+function AbstractAlgebra.expressify(@nospecialize(a::AffineCurveDivisor); context = nothing)
+   prod = Expr(:call, :+)
+   for (p, i) in a.divisor
+      ep = AbstractAlgebra.expressify(p.coord, context = context)
+      push!(prod.args, Expr(:call, :*, i, ep))
+   end
+   return prod
+end
+
+function AbstractAlgebra.needs_parentheses(A::Oscar.Geometry.ProjSpcElem{T}) where T <: RingElem
+    return false
+end
+
+function AbstractAlgebra.expressify(@nospecialize(a::ProjCurveDivisor); context = nothing)
+   prod = Expr(:call, :+)
+   for (p, i) in a.divisor
+      ep = AbstractAlgebra.expressify(p, context = context)
+      push!(prod.args, Expr(:call, :*, i, ep))
+   end
+   return prod
+end
+
+function Base.show(io::IO, ::MIME"text/plain", a::CurveDivisor)
+  print(io, AbstractAlgebra.obj_to_string(a, context = io))
+end
+
+function Base.show(io::IO, a::CurveDivisor)
+  print(io, AbstractAlgebra.obj_to_string(a, context = io))
+end
+
+################################################################################
+################################################################################
+
+function help_add(D::CurveDivisor, E::CurveDivisor)
+    D.C == E.C || error("The divisors are not on the same curve.")
+    F = copy(D.divisor)
+    G = copy(E.divisor)
+    ks = keys(F)
+    for P in keys(E.divisor)
+        if P in ks
+            a = F[P] + G[P]
+            F[P] = a
+            delete!(G, P)
+        end
+    end
+    for P in keys(G)
+        F[P] = G[P]
+    end
+    return F
+end
+
+################################################################################
+# Sum of two divisors
+
+function Base.:+(D::AffineCurveDivisor, E::AffineCurveDivisor)
+    F = help_add(D, E)
+    return AffineCurveDivisor(D.C, F)
+end
+
+function Base.:+(D::ProjCurveDivisor, E::ProjCurveDivisor)
+    F = help_add(D, E)
+    return ProjCurveDivisor(D.C, F)
+end
+
+################################################################################
+
+function Base.:-(D::CurveDivisor, E::CurveDivisor)
+    return D + -1*E
+end
+
+################################################################################
+################################################################################
+
+function help_int_multiplication(D::CurveDivisor, k::Int)
+    F = copy(D.divisor)
+    for P in keys(F)
+        F[P] = k*D.divisor[P]
+    end
+    return F
+end
+
+function Base.:*(k::Int, D::AffineCurveDivisor)
+    F = help_int_multiplication(D, k)
+    return AffineCurveDivisor(D.C, F)
+end
+
+function Base.:*(k::Int, D::ProjCurveDivisor)
+    F = help_int_multiplication(D, k)
+    return ProjCurveDivisor(D.C, F)
+end
```

Comment:
Perhaps code can be unified as above? Also, might make sense to special case k==0 and k==1?